app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -71,28 +128,164 @@ class PopupView : FrameLayout {
     }
 
     /**
-     * Iterates over all the display fields of the popup to get their values and adds them to the
-     * list of PopupAttributes.
+     * Updates and saves popup field values that have been edited by the user.
      */
-    private fun getPopupAttributeList(): ArrayList<PopupAttribute> {
-        val popupManager = PopupManager(context, popup)
-        val fields: List<PopupField> = popupManager.displayedFields
-        val popupDisplayFieldsWithValues: ArrayList<PopupAttribute> = ArrayList()
-        for (field in fields) {
-            popupDisplayFieldsWithValues.add(
-                PopupAttribute(
-                    field.label,
-                    popupManager.getFormattedValue(field)
+    private fun saveEdits() {
+        popupManager.startEditing()
+        var exception: ArcGISRuntimeException?
+        for (attribute in popupAttributeList) {
+            if (attribute.tempValue != null && attribute.tempValue.toString() != attribute.value && attribute.isEditable) {
+                // update the edits locally
+                exception =
+                    updateValue(popupManager, attribute.field, attribute.tempValue.toString())
+                if (exception != null) {
+                    val fieldName: String = attribute.fieldName
+                    val message =
+                        "There was an error saving the feature, the value for  $fieldName is invalid"
+                    showSnackbarMessage(message)
+                } else {
+                    // update the attribute value with the new value
+                    attribute.value = attribute.tempValue.toString()
+                    savePopup()
+                }
+            }
+        }
+    }
+
+    /**
+     * Shows the message in a Snackbar.
+     */
+    private fun showSnackbarMessage(message: String) {
+        val snackbar: Snackbar = Snackbar.make(this, message, Snackbar.LENGTH_LONG)
+        snackbar.show()
```

Comment:
This can be shortened?
```suggestion
        Snackbar.make(this, message, Snackbar.LENGTH_LONG).show()
```

Reply:
updated